### ISSUE TITLE
Interpretation of glTF framerate breaks exported animations from Blender 3.6

### ISSUE THREAD

**Issue by izzyboris** [Moguri/panda3d-gltf#127]
This was sparked by the discussion here: https://github.com/Moguri/blend2bam/issues/84

I dug into this some more after that fix to blend2bam went in, and animations still didn't work for me.

Blender 3.3 would write the number of frames as 'count' on the glTF accessor, but in 3.6 this value seems to always be '2'. I believe Blender's interpretation of 'count' changed from "number of animation frames" to "number of time values" ([start, end[), but that's just a guess since my understanding of the usage of 'count' from this spec doesn't make it very clear: https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#accessors-overview

This post additionally discusses the fact that glTF standardizes on exporting frame timing by seconds, not framerate: https://github.com/KhronosGroup/glTF/issues/1408 and that there's no "standard framerate" for glTF, only an animation time, which I believe means the glTF converter has to assume a framerate (24) and infer the number of frames itself when converting to a Panda3d AnimBundle.

Something as simple as this in Converter.build_character() fixes exported blender animations for me, assuming 24 FPS and calculating the number of frames, inverting the current logic:

```diff
-            num_frames = time_acc['count']
+            # glTF records frame time in seconds, not by framerate,
+            # so calculate frame count assuming a 24-fps playback
             end_time = time_data[-1]
-            fps = num_frames / time_data[-1] if end_time != 0 else 24
+            fps = 24
+            num_frames = int(fps * end_time if end_time != 0 else 0)
```

However, I worry there may be pitfalls here. It should be fine from an animation perspective if someone animates at a different framerate, because the glTF spec normalizes it by animation time. But the calculation *has* to work out -- if it somehow (unexpectedly) ends up with a non-integer number of frames, either we truncate and drop a frame or round up and things look off.

So does this look like the right fix?

**Comment by Moguri** [comment]
I do not believe this is quite the right fix. However, it is going in the right direction. I've always known that this logic was pretty Blender specific (and largely why sampling had to be selected when exporting). As you note, glTF doesn't care about an fps value, but Panda is more frame/fps-based for it's animation API. So, we need to find an appropriate conversation that doesn't result in dropped frames (otherwise looping will break). I am sure this is a well-solved problem, but researching it has not been a priority for me. Now that it's causing more problems I have more incentive to fix it.

**Comment by Moguri** [comment]
This is fixed by 3214c3bacf19ca9f8098352e9878d5777d1ab65b
